### DISCUSSION TITLE
Monitoring Interface for Medusa

### DISCUSSION THREAD

**Discussion by annomatik** [pymedusa/Medusa#5115] [Ideas & Feature requests]
**Is your feature request related to a problem? Please describe.**
I have a lot of different apps and would like to see their status in one place. That's why I have installed Pandora FMS as a monitoring solution. I currently check, if my local Medusa website can be reached and if the process is running, but I'd like to do a deeper monitoring

**Describe the solution you'd like**
I'd like to have access to "Server Status" of Medusa + the currently uncleared errors + warning (both as numbers and a list, maybe even with links into the Medusa log list)

- Are the schedulers alive (one line per Scheduler)? (if not --> problem)
- How much disk space is left, for both TV Download directory and Media Root directory (I can do that from outside)?
- How many active shows are there (if 0 --> problem)
- How long does the daily search take, when was the last daily search (if too long ago --> problem)
- how long does the backlog search take, when was the last backlog search (if too long ago --> problem)
- how long does the loading of the show overview page take (if too long --> problem)
- how much memory does a call to the Medusa Website cost
- how much CPU time does a call to the Medusa Website cost
- list of current (=not cleared) errors
- list of current (=not cleared) warnings

...everything, if possible, delivered in a XML or some sort of CSV stream I can process with a bash script.

Thanks!

**Describe alternatives you've considered**
I'm currently monitoring, if the Web-Site is reachable and if the process is running, but I'd like to go deeper.

**Additional context**
Thanks!

**Comment by p0psicles** [comment]
You can already get most of the information just using the web and api calls

**Comment by annomatik** [comment]
Cool. Do you have a link to the corresponding documentation for me?

**Comment by p0psicles** [comment]
Just check the network tab in chrome. That should give you enough to get started on

**Comment by annomatik** [comment]
Well... I looked into the network tab, there's apparently over 30 different JavaScripts and to be honest, I have no idea, where to look for the info I'm looking for. I can invest a couple of hours, try to reverse-engineer everything but some sort of pointer to where to start would be nice. Also, an actual documented API, as small as it might be, would have the advantage of being less guess-work and more stable ( I hope :-) )

**Comment by p0psicles** [comment]
For the status overview, you'll have to scrape the site for now.
`medusa/home/status/` will just give you html. I thought we already moved this to apiv2. But that's not 'yet' the case.

For the logs, here you can check on the api, used to get logs:
https://github.com/pymedusa/Medusa/blob/develop/dredd/api-description.yml

You can use https://editor.swagger.io/ and past the contents for api-description.yml there, to get a more graphical overview.

**Comment by p0psicles** [comment]
Fastest way to check whats available to you, is by using chrome developer tools, and go from what's in the network tab. Next is the api-description.yml. And then you also have the 'old' apibuilder (apiv1). http://yourmedusa.url:8881/apibuilder

**Comment by p0psicles** [comment]
If I where you, I would use python, to create small scripts, that login to medusa. And scrape (for now) the status page. And other scripts that utilize the apiv1 and apiv2. Here is some example code, I used for the Kodi addon. https://github.com/pymedusa/kodi-failed-downloads-addon/blob/master/resources/lib/context.py#L45

**Comment by p0psicles** [comment]
Also where in the transitioning of having a frontend based on prerendered html, towards having everything available in the apiv2. So in the coming releases you will see that more and more will become available in the apiv2. And with that the api-description.yml will be extended.

**Comment by OmgImAlexis** [comment]
@p0psicles what about adding something like https://github.com/cyberdelia/metrology we could export all sorts of metrics for users. I'd **LOVE** a stats page. 🙏

**Comment by p0psicles** [comment]
Yeah looks cool

**Comment by annomatik** [comment]
p0psicles, Thanks, looking forward to the apis and will try to scrape the status page for now.
OmgImAlexis, Cool idea!

Thank you both!
